sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java

```diff
@@ -371,54 +374,108 @@ public DataLakeServiceVersion getServiceVersion() {
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure Docs</a></p>
      *
-     * @param accessControl {@link PathAccessControl}
+     * @param accessControlList The access control list of the resource.
+     * @param group The group of the resource.
+     * @param owner The owner of the resource.
      * @return A reactive response containing the resource info.
      */
-    public Mono<PathInfo> setAccessControl(PathAccessControl accessControl) {
+    public Mono<PathInfo> setAccessControlList(List<PathAccessControlEntry> accessControlList, String group,
+        String owner) {
         try {
-            return setAccessControlWithResponse(accessControl, null).flatMap(FluxUtil::toMono);
+            return setAccessControlListWithResponse(accessControlList, group, owner, null).flatMap(FluxUtil::toMono);
         } catch (RuntimeException ex) {
             return monoError(logger, ex);
         }
+    }
 
+    /**
+     * Changes the access control list for a resource.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakePathAsyncClient.setAccessControlListWithResponse#List-String-String-DataLakeRequestConditions}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure Docs</a></p>
+     *
+     * @param accessControlList The access control list of the resource.
+     * @param group The group of the resource.
+     * @param owner The owner of the resource.
+     * @param accessConditions {@link DataLakeRequestConditions}
+     * @return A reactive response containing the resource info.
+     */
+    public Mono<Response<PathInfo>> setAccessControlListWithResponse(List<PathAccessControlEntry> accessControlList,
+        String group, String owner, DataLakeRequestConditions accessConditions) {
+        try {
+            return withContext(context -> setAccessControlWithResponse(accessControlList, null, group, owner,
+                accessConditions, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
     }
 
     /**
-     * Changes the access control for a resource.
+     * Changes the permissions for a resource.
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.storage.file.datalake.DataLakePathAsyncClient.setAccessControlWithResponse#PathAccessControl-DataLakeRequestConditions}
+     * {@codesnippet com.azure.storage.file.datalake.DataLakePathAsyncClient.setPermissions#PathPermissions-String-String}
      *
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure Docs</a></p>
      *
-     * @param accessControl {@link PathAccessControl}
+     * @param permissions The permissions of the resource.
```

Comment:
I tend to try to just have the docstrings for parameters like this be just {@link PathPermissions} and let that redirect to the docs on the type. Then we don't have to worry about keeping the docs consistent everywhere the type is used. Not sure how others feel about it, though.

Reply:
good point. done